homeassistant/components/hvac/__init__.py

```diff
@@ -481,13 +484,13 @@ def max_humidity(self):
 
     def _convert_for_display(self, temp):
         """Convert temperature into preferred units for display purposes."""
-        if temp is None:
-            return None
+        if temp is None or not isinstance(temp, Number):
```

Comment:
I'd prefer raise error over no-conversion - but again, trying to follow existing logic. It sounds like I should break existing logic and just raise an error if anything goes wrong/unexpected.